### PR TITLE
Make ScriptLog.GetOutput async in preparation for the Kubernetes Agent implementation

### PR DESCRIPTION
**Pull request by flin-8**
[sc-71685]

# Background

The Kubernetes Agent will read logs from the Kubernetes API, which are `async` calls. 

This PR makes the signature `async` to allow the change. Its callers are already async, so there shouldn't be any problems here.
# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests.Integration/Util/TestScriptLog.cs [hunk 1]
```diff
@@ -15,7 +16,7 @@ public class TestScriptLog : IScriptLog, IScriptLogWriter
         public IScriptLogWriter CreateWriter()
             => this;
 
-        public List<ProcessOutput> GetOutput(long afterSequenceNumber, out long nextSequenceNumber)
+        public Task<(List<ProcessOutput> Outputs, long NextSequenceNumber)> GetOutput(long afterSequenceNumber)
```
Comment (flin-8) [inline source/Octopus.Tentacle.Tests.Integration/Util/TestScriptLog.cs:19]: `async` functions can't have `out` parameters - return as part of the tuple

**Review by scme0 — APPROVED** [review]
LGTM

**Review by rhysparry — APPROVED** [review]
LGTM

**Comment by flin-8** [comment]
I might not need this, depending on how we refactor the script services. Will keep you updated

**Pull request closed without merging**